src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -486,4 +300,79 @@ impl HummockManager {
             }
         }
     }
+
+    async fn correct_commit_ssts(
+        &self,
+        sstables: Vec<LocalSstableInfo>,
+        table_compaction_group_mapping: &HashMap<TableId, CompactionGroupId>,
+    ) -> Result<BTreeMap<CompactionGroupId, Vec<SstableInfo>>> {
+        let mut new_sst_id_number = 0;
+        let mut sst_to_cg_vec = Vec::with_capacity(sstables.len());
+        for commit_sst in sstables {
+            let mut group_table_ids: BTreeMap<u64, Vec<u32>> = BTreeMap::new();
+            for table_id in commit_sst.sst_info.get_table_ids() {
+                match table_compaction_group_mapping.get(&TableId::new(*table_id)) {
+                    Some(cg_id_from_meta) => {
+                        group_table_ids
+                            .entry(*cg_id_from_meta)
+                            .or_default()
+                            .push(*table_id);
+                    }
+                    None => {
+                        tracing::warn!(
+                            "table {} in SST {} doesn't belong to any compaction group",
+                            table_id,
+                            commit_sst.sst_info.get_object_id(),
+                        );
+                    }
+                }
+            }
+
+            new_sst_id_number += group_table_ids.len();
+            sst_to_cg_vec.push((commit_sst, group_table_ids));
+        }
+
+        // Generate new SST IDs for each compaction group
+        // `next_sstable_object_id` will update the global SST ID and reserve the new SST IDs
+        // So we need to get the new SST ID first and then split the SSTs
+        let mut new_sst_id = next_sstable_object_id(&self.env, new_sst_id_number).await?;
```

Comment:
May add a sanity check that the `new_sst_id` won't be incremented to larger than the max-prefetched `sst_id`